### DISCUSSION TITLE
reactive-aggregate & redis-oplog

### DISCUSSION THREAD

**Discussion by DeanKamali** [robfallows/tunguska-reactive-aggregate#51] [Q&A]
I'm wondering if `tunguska-reactive-aggregate` can work in connection with `redis-oplog`  
https://github.com/cult-of-coders/redis-oplog

Adding `redis-oplog` to my application broke all reactivity, I didn't really spend much time investigating this as I reverted the change, but the first thing that came to my mind was something is not compatible with `tunguska-reactive-aggregate`

Any ideas?

**Comment by robfallows** [comment]
Good question! The short answer is I don't know.

It's definitely something that I'd want to work if at all possible. I'm a bit pushed for free time at the moment, but I'll definitely look into this. Obviously, if you come across any further information, please add to this topic.

Thank you 👍

**Reply by laidinidis** [reply]
We're using `redis-oplog` and recently i tried `tunguska-reactive-aggregate`. It looks like it works fine (local and staging), my only issue is that it takes some time to initialize the observers https://github.com/robfallows/tunguska-reactive-aggregate/discussions/53

**Reply by robfallows** [reply]
Yep. I can confirm it works. Obviously it has the usual redis-oplog issue with out-of-band mutations, but otherwise seems ok.

**Reply by DeanKamali** [reply]
please help me understand this, in my aggregations, I do have observers on many of them, so you guys are saying it will take some time to load?
I have reviewed #52,  I'm a little confused about how to add Meteor.setTimeout to the mix

**Reply by robfallows** [reply]
I don't know whether you'll be affected. It probably depends on the complexity and number of observers. If you've noticed nothing to date, I don't think it will affect you.

However, see #52 discussion. I've made a testing branch for the changes to asynchronously set up the observers. If you're interested/concerned please follow the testing instructions in there to see if it makes any difference.
